src/components/input/input.tsx

```diff
@@ -434,7 +434,10 @@ export class Input implements LabelableComponent, FormComponent {
 
   private inputBlurHandler = () => {
     if (this.type === "number") {
-      this.setValue({ value: this.value });
+      this.setValue({
+        value: this.value,
+        origin: "external"
+      });
```

Comment:
Using `origin: "external"` here distorts the meaning and intention behind the `external` label which is intended to only apply to programmatic changes to the value originating from external javascript, and not as a result of a user interacting with the input normally.  Because of this, the only place where `external` can be used is from within the `valueWatcher` and ONLY when `this.internalValueChange` is set to `false`.